File: typescript/solution.test.ts
import { exec } from 'child_process';
import { promisify } from 'util';

const execAsync = promisify(exec);

describe('solution.ts output', () => {
  it('should print the correct output to the console', async () => {
    const { stdout } = await execAsync('ts-node solution.ts');

    const expected = 'HIPPOPOTOMONSTROSESQUIPPEDALIOPHOBIA';
    expect(stdout.trim()).toBe(expected);
  });
});

File: ruby/solution.test.rb
require 'rspec'

RSpec.describe 'Playfair Decryption Script' do
  it 'outputs the correct decrypted text' do
    output = `ruby solution.rb` # Using backticks to execute the script and capture the output.
    expect(output.strip).to eq('HIPPOPOTOMONSTROSESQUIPPEDALIOPHOBIA') # Compares the stripped output to the expected string.
  end
end
